libraries/core-react/src/Button/Button.jsx

```diff
@@ -105,36 +100,37 @@ const ButtonBase = styled.button.attrs(({ type = 'button' }) => ({
   }
 `
 
-// TODO:
-// - Missing top spacing
-// - Fallback color?
-// - Use ThemeProvider
-// - Set attr on button
-
-export const Button = ({
-  variant,
-  children,
-  disabled,
-  className,
-  color,
-  ...other
-}) => {
+export const Button = forwardRef(function Button(
+  { variant, children, disabled, className, color, tabIndex, href, ...other },
+  ref,
+) {
   const colorBase = colors[color] || {}
   const base = colorBase[variant] || {}
   const baseDisabled = colors.disabled[variant] || {}
 
+  const as = href ? 'a' : other.as ? other.as : 'button'
+
+  const baseProps = {
+    ...other,
+    ref,
+    as,
+  }
+
   return (
     <ButtonBase
       base={base}
+      type={href || other.as ? undefined : 'button'}
```

Comment:
We should maybe not ignore `type` if it comes from props either. Force it if href is put, else use provided type?